### ISSUE TITLE
RabbitMQ missing features

### ISSUE THREAD

**Issue by johnbfair** [colinbull/FSharpEnt#6]
I'd like to implement a few features (at least for PubSub) that are missing from the RabbitMQ client. For instance:
- Queue Durability - This is currently defaulted to false, but needs to be configurable by the end user
- Confirms - The user should be able to decide if they want to leverage confirms or not
  `
  model.ConfirmSelect();
  model.Value.WaitForConfirmsOrDie(timeout) 
  `
- Persistence - The user should be able to decide if they want to leverage persistence as well
  `
  model.CreateBasicProperties(DeliveryMode=2uy)
  ` 
- BasicAck/BasicNack - These events aren't implemented (which are integral for pieces like Confirms)

That being said...if you have any suggestions for how to accomplish all of that (making changes to Location to account for some new fields, etc) let me know.  

Once we get to feature parity w/ the (much much smaller) library that I made for testing [F# Agents w/ RabbitMQ](https://github.com/johnbfair/FSharp-and-RabbitMQ) then I'll re-run my performance tests (Console app + Ants perf profiler) and see if anything else jumps out.

**Comment by colinbull** [comment]
Yes, I agree all of these things should be added. At the time of writing I had a very small task I need to accomplish so this covers that and nothing else. 

As for the implementation maybe create a queue DU something like 

```
 type Queue = 
     | Queue of string 
     | QueueWithProps of string * QueueProperties
```

and then changing the Location type to suit. The same type of extension could be done for exchanges I guess aswell. 

As for F# agents and stuff, on the [F# actors develop branch](https://github.com/colinbull/Fsharp.Actor/tree/develop/src/FSharp.Actor) I will hopefully include a rabbitMQ transport for this. Alas it is all about time currently as I appear to have spread myself a little thinly and hence my opensource time has somewhat slipped.  

Cheers

**Comment by johnbfair** [comment]
Sounds good.  I can certainly pitch in and help out on this piece.  If I mess anything up just let me know. :wink:  I'll definitely look to go the Queue DU route as well.  :thumbsup:
